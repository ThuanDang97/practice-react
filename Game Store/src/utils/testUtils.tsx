import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import { BrowserRouter as Router } from 'react-router-dom'

// Themes
import theme from '@self-types/Themes'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <AuthProvider>
        <CartProvider>
          <Router>{component}</Router>
        </CartProvider>
      </AuthProvider>
    </ThemeProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
