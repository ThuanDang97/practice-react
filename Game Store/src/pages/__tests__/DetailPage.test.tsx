// Utils
import { render } from '@utils/testUtils'

// Page
import DetailPage from '@pages/detail'

describe('Detail Page', () => {
  it('should render Detail Page', () => {
    const { container } = render(<DetailPage />)

    expect(container).toMatchSnapshot()
  })
})
