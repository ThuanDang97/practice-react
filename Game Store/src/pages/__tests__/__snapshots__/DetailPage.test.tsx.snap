// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Page should render Detail Page 1`] = `
<div>
  <div
    class="main MuiBox-root css-ejp65w"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-dl3eaz-MuiTypography-root"
    >
      Loading...
    </h1>
  </div>
</div>
`;
